Update/start screen

devzl opened the pull request:
## Description

Please include a summary of the change and which issue is fixed. 
Update start screen UI
Fixes #1616 

## Which feature is affected?
- [ ] Create vault ( Secure / Fast) - Please ensure you created a Secure vault & fast vault
- [ ] Sending  - Please attach a tx link here
- [ ] Swap - Please attach a tx link for the swap here
- [ ] New Chain / Chain related feature  -  Please attach tx link here

## Checklist

- [ ] I have performed a self-review of my code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works

## Screenshots (if applicable):
<img width="407" alt="Screenshot 2025-05-21 at 12 03 24" src="https://github.com/user-attachments/assets/ca1b086e-d563-4cbe-920c-03f92bd886d6" />


## Additional context

<!-- This is an auto-generated comment: release notes by coderabbit.ai -->
## Summary by CodeRabbit

- **New Features**
  - Buttons now support displaying an icon alongside their text.
  - Conditional display of back button in new vault page based on vault presence.

- **Style**
  - Updated button and icon button styling for improved consistency and appearance across pages.
  - Adjusted button sizes, padding, and color schemes for a more unified look.
  - Reduced the font size of the product logo text.
  - Added a minimum height to page headers for better layout stability.
  - Enhanced spacing and alignment in various UI components, including added horizontal padding in vault connected app section.

- **Refactor**
  - Replaced older button components with standardized versions from a centralized UI library.
  - Updated button properties to align with new UI conventions.
  - Switched several buttons to icon buttons with updated sizing and styling.

- **Bug Fixes**
  - Ensured consistent rendering of page headers and controls based on context (e.g., vault presence).
<!-- end of auto-generated comment: release notes by coderabbit.ai -->

rcoderdev commented on core/ui/vault/new/index.tsx:
```diff
@@ -1,40 +1,54 @@
     <VStack fullHeight>
+      {vaults.length > 0 && (
+        <PageHeader
+          style={{ height: '55px' }}
```
Why is there a hardcoded height here? I believe the page header should maintain the same height across all screens.

devzl replied:
The PageHeader component only applies 16px of padding, and when it's used without the title property (like in our case here) it loses height consistency with the rest of the app. 
Another possible solution would be to set height on the Container wrapper inside the PageHeader component
<img width="401" alt="Screenshot 2025-05-21 at 13 22 16" src="https://github.com/user-attachments/assets/8d919363-b203-4d14-97fb-b1dadbacd3bf" />

rcoderdev commented on core/ui/vault/new/index.tsx:
```diff
@@ -1,40 +1,54 @@
           </Button>
           <Button
-            kind="secondary"
+            rounded
```
Please double-check to ensure that all our buttons are round.?

devzl replied:
Yes, all 3 start screen buttons are rounded

Ehsan-saradar reviewed (changes requested):
I noticed there are two different button in the lib. I think we only need one. We should either update the old button with the new style and remove the new one from the extension, or replace the old button with the new one entirely.

What do you think, @rcoderdev ?

devzl commented:
@Ehsan-saradar @rcoderdev I removed the Button component file from the extension client and updated the Button component in the lib folder with the new design, as suggested above in my last commit

rcoderdev commented on lib/ui/theme/getHoverVariant.ts:
```diff
@@ -22,4 +28,7 @@ export const getHoverVariant =
         theme.colors.primary.getVariant({ l: (l: number) => l * 0.85 }),
       foregroundSuperContrast: () => theme.colors.foregroundSuperContrast,
       idle: () => theme.colors.idle.getVariant({ l: (l: number) => l + 4 }),
+      buttonPrimaryWebHover: () => theme.colors.buttonPrimaryWebHover,
```
Can we update our existing colors instead of introducing a buttonPrimaryWeb? What does "web" mean in this context?

rcoderdev replied:
Why does it need a hover warning if it already has the 'hover' suffix?

devzl replied:
I used existing ones

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -156,8 +203,12 @@ type ButtonProps = Omit<
   isLoading?: boolean
   isRounded?: boolean
   kind?: ButtonKind
+  isBlock?: boolean
```
We don't need the `isBlock` property, as I mentioned earlier. Instead, we can either make the container stretch the button by setting its width to 100% of the parent component, or, if necessary, we can implement a `fullWidth` prop like we do in other components, such as Stack.

devzl replied:
Removed

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -23,13 +26,20 @@ type ButtonKind =
   kind: ButtonKind
+  fitContent?: boolean
+  status?: Status
+  weight?: Weight
```
Weight doesn't make sense; it should be encapsulated by kind. Therefore, weight should be based on kind.

devzl replied:
Removed

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -8,13 +8,16 @@ import { getHoverVariant } from '@lib/ui/theme/getHoverVariant'
 import { UnstyledButton } from './UnstyledButton'
 
 type ButtonSize = 'xs' | 's' | 'm' | 'l' | 'xl'
+type Status = 'default' | 'error' | 'success' | 'warning'
```
We already have the `kind` and `isPending` | `error` fields. Why introduce new fields that conflict with the existing ones?

devzl replied:
Removed

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -8,13 +8,16 @@ import { getHoverVariant } from '@lib/ui/theme/getHoverVariant'
 
 type ButtonSize = 'xs' | 's' | 'm' | 'l' | 'xl'
+type Status = 'default' | 'error' | 'success' | 'warning'
+type ButtonType = 'default' | 'link'
```
We already have a `kind` prop. I believe we don't need to add any new props to the button component. Instead, we can update the existing `kind`-based styles  to align with the new deskign if necessary. Alternatively, we can leave it as is and create a new ticket to update the buttons to the new UI standard. Every prop or parameter should have a clear purpose and should not conflict with existing ones.

devzl replied:
Now using only `kind`

rcoderdev reviewed (approved):
Double-check Icon prop, and free to merge

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -23,13 +24,17 @@ type ButtonKind =
   isDisabled?: boolean
   isLoading?: boolean
   isRounded?: boolean
+  icon?: ReactNode
```
Might be safer to allow the user of the component to decide how they want to position an icon relative to the text, especially since we don't have a design system in place.

devzl commented:
@rcoderdev Thanks for the feedback!
For now, this PR updates the start screen as intended, and I’ve updated the lib/button component based on the comment above to support the existing logic.
If we want to make further improvements related to icon handling for future use cases, I suggest doing that in a separate ticket. The more extra changes or delays we introduce here, the more files end up getting conflicted since this button is used across the app.

Ehsan-saradar reviewed (changes requested):
@devzl 
Previously, all buttons were rounded by default, but with the recent change, rounding is now an optional property. Please ensure that all buttons that were previously rounded explicitly include the rounded prop to maintain their intended appearance.

Also horizontal padding was removed, but all buttons should maintain their original spacing. Please ensure that each button has the same horizontal padding as before to preserve consistency and layout.

![telegram-cloud-photo-size-4-5888464914199202259-x](https://github.com/user-attachments/assets/15c62d90-dcdd-4dbe-9fc6-7e737cfa5aa7)

rcoderdev commented:
> @rcoderdev Thanks for the feedback! For now, this PR updates the start screen as intended, and I’ve updated the lib/button component based on the comment above to support the existing logic. If we want to make further improvements related to icon handling for future use cases, I suggest doing that in a separate ticket. The more extra changes or delays we introduce here, the more files end up getting conflicted since this button is used across the app.

No additional changes are needed; you can simply remove the `icon` prop from the Button, as it was before. Additionally, as Ehsan mentioned, I don't believe we have any buttons that are not rounded, so you can eliminate the `isRounded` prop entirely.

devzl commented:
> @devzl Previously, all buttons were rounded by default, but with the recent change, rounding is now an optional property. Please ensure that all buttons that were previously rounded explicitly include the rounded prop to maintain their intended appearance.
> 
> Also horizontal padding was removed, but all buttons should maintain their original spacing. Please ensure that each button has the same horizontal padding as before to preserve consistency and layout.
> 
> ![telegram-cloud-photo-size-4-5888464914199202259-x](https://private-user-images.githubusercontent.com/7792430/448402489-15c62d90-dcdd-4dbe-9fc6-7e737cfa5aa7.jpg?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3NDg0NjE2MDYsIm5iZiI6MTc0ODQ2MTMwNiwicGF0aCI6Ii83NzkyNDMwLzQ0ODQwMjQ4OS0xNWM2MmQ5MC1kY2RkLTRkYmUtOWZjNi03ZTczN2NmYTVhYTcuanBnP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI1MDUyOCUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNTA1MjhUMTk0MTQ2WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9YjM4N2U2ZWU1ZTk0ZjczNDI1YTc2OWUwNjc0NmM3MGNkYTFhZTVjM2UyYmUyMTM0MjFkOGU4ZjU3OTE4NmM5MSZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.T8mD4V9GupUDArCiMfatEQmshWsIa-XKxfHn62cW8qg)

<img width="403" alt="Screenshot 2025-05-28 at 20 39 46" src="https://github.com/user-attachments/assets/81b3d327-2055-440e-ac3a-583b2e2a5dc5" />
@Ehsan-saradar I added padding for buttons and made them rounded

rcoderdev commented on lib/ui/buttons/Button.tsx:
```diff
@@ -154,42 +160,50 @@ type ButtonProps = Omit<
     </>
   ) : (
-    children
+    <HStack gap={8} alignItems="center">
```
If you retain the icon prop, wrap it with HStack only when the icon is provided. For example, you could use something like this:

```javascript
const innerContent = icon ? <HStack gap={8} alignItems="center">{icon}{children}</HStack> : children
```

Then, use this value for both the loading state and the regular state.

Ehsan-saradar commented:
@rcoderdev 
Also, I noticed that many of the good features from the new button have been discarded, and we're reverting to the old button. As a result, the UI of the extension no longer matches the Figma design, for example, in the connected dApp pages.
I suggest splitting this PR into two separate pull requests:
1- Fix the issue that is necessary for release: If the vault already exists, enable the back arrow to allow users to return to the Vault selection.
2- Unify the buttons and merge them into a single component. I can take care of this part if you'd like.
Let me know your thoughts @devzl

Before
Check the disconnect all button and also the primary color for the disconnect button
![telegram-cloud-photo-size-4-5890716714012887321-x](https://github.com/user-attachments/assets/14bc98f1-4bb5-403c-a10e-4f0318ca5ea6)
Check hover style
![telegram-cloud-photo-size-4-5890716714012887322-x](https://github.com/user-attachments/assets/6db7c051-4cc7-41c8-bf0e-4e9f43642c84)

After
Compare the disconnect all button and also primary buttons
![telegram-cloud-photo-size-4-5890716714012887320-x](https://github.com/user-attachments/assets/0c398e15-14ff-4a4e-a8a2-b8ecf72e819a)


Or loading style

Ehsan-saradar commented:
Also, compare loading, disable 
![telegram-cloud-photo-size-4-5890716714012887352-x](https://github.com/user-attachments/assets/047a9bbf-b755-4bb4-a217-b46aad6b0c13)
